Sources/Sentry/SentryNetworkTrackingIntegration.m

```diff
@@ -10,45 +10,49 @@ @implementation SentryNetworkTrackingIntegration
 
 - (void)installWithOptions:(SentryOptions *)options
 {
-    // We are aware that the SDK only creates breadcrumbs for HTTP requests if performance is
-    // enabled. A proper fix is not straight forward as we need several checks on SentryOptions in
-    // SentryNetworkTracker. As we have a problem with KVO, see
-    // https://github.com/getsentry/sentry-cocoa/issues/1328, we don't know if we can keep the
-    // SentryNetworkTracker (written on 29th of September 2021). Therefore we accept this tradeof
-    // for now.
+    if (!options.enableSwizzling) {
+        [SentryLog logWithMessage:
+                       @"Not going to enable NetworkTracking because enableSwizzling is disabled."
+                         andLevel:kSentryLevelDebug];
+        return;
+    }
+
+    BOOL needNetworkTracking = true;
 
     if (!options.isTracingEnabled) {
         [SentryLog logWithMessage:
                        @"Not going to enable NetworkTracking because isTracingEnabled is disabled."
                          andLevel:kSentryLevelDebug];
-        return;
+        needNetworkTracking = false;
     }
 
-    if (!options.enableAutoPerformanceTracking) {
+    if (needNetworkTracking && !options.enableAutoPerformanceTracking) {
         [SentryLog logWithMessage:@"Not going to enable NetworkTracking because "
                                   @"enableAutoPerformanceTracking is disabled."
                          andLevel:kSentryLevelDebug];
-        return;
+        needNetworkTracking = false;
     }
 
-    if (!options.enableNetworkTracking) {
+    if (needNetworkTracking && !options.enableNetworkTracking) {
         [SentryLog
             logWithMessage:
                 @"Not going to enable NetworkTracking because enableNetworkTracking is disabled."
                   andLevel:kSentryLevelDebug];
-        return;
+        needNetworkTracking = false;
     }
 
-    if (!options.enableSwizzling) {
-        [SentryLog logWithMessage:
-                       @"Not going to enable NetworkTracking because enableSwizzling is disabled."
-                         andLevel:kSentryLevelDebug];
-        return;
+    if (needNetworkTracking) {
+        [SentryNetworkTracker.sharedInstance enableNetworkTracking];
+        [SentryNetworkTrackingIntegration swizzleNSURLSessionConfiguration];
+    }
+
+    if (options.enableNetworkBreadcrumbs) {
+        [SentryNetworkTracker.sharedInstance enableBreadcrumbs];
     }
 
-    [SentryNetworkTracker.sharedInstance enable];
-    [SentryNetworkTrackingIntegration swizzleNSURLSessionConfiguration];
-    [SentryNetworkTrackingIntegration swizzleURLSessionTask];
+    if (needNetworkTracking || options.enableNetworkBreadcrumbs) {
+        [SentryNetworkTrackingIntegration swizzleURLSessionTask];
```

Comment:
`m`: I just realized that commenting out this line doesn't lead to a failing test in `SentryNetworkTrackerIntegrationTests`. Most likely this is my fault, but could you be so kind and add a test for this if possible, please?

Reply:
Sure.

Reply:
Sure